providers/ms365/resources/ms365_exchange.go

```diff
@@ -85,28 +85,41 @@ ConvertTo-Json -Depth 4 $exchangeOnline
 `
 
 type ExchangeOnlineReport struct {
-	MalwareFilterPolicy            []interface{}     `json:"MalwareFilterPolicy"`
-	HostedOutboundSpamFilterPolicy []interface{}     `json:"HostedOutboundSpamFilterPolicy"`
-	TransportRule                  []interface{}     `json:"TransportRule"`
-	RemoteDomain                   []interface{}     `json:"RemoteDomain"`
-	SafeLinksPolicy                []interface{}     `json:"SafeLinksPolicy"`
-	SafeAttachmentPolicy           []interface{}     `json:"SafeAttachmentPolicy"`
-	OrganizationConfig             interface{}       `json:"OrganizationConfig"`
-	AuthenticationPolicy           interface{}       `json:"AuthenticationPolicy"`
-	AntiPhishPolicy                []interface{}     `json:"AntiPhishPolicy"`
-	DkimSigningConfig              interface{}       `json:"DkimSigningConfig"`
-	OwaMailboxPolicy               interface{}       `json:"OwaMailboxPolicy"`
-	AdminAuditLogConfig            interface{}       `json:"AdminAuditLogConfig"`
-	PhishFilterPolicy              []interface{}     `json:"PhishFilterPolicy"`
-	Mailbox                        []interface{}     `json:"Mailbox"`
-	AtpPolicyForO365               []interface{}     `json:"AtpPolicyForO365"`
-	SharingPolicy                  []interface{}     `json:"SharingPolicy"`
-	RoleAssignmentPolicy           []interface{}     `json:"RoleAssignmentPolicy"`
-	ExternalInOutlook              []*ExternalSender `json:"ExternalInOutlook"`
+	MalwareFilterPolicy            []interface{} `json:"MalwareFilterPolicy"`
+	HostedOutboundSpamFilterPolicy []interface{} `json:"HostedOutboundSpamFilterPolicy"`
+	TransportRule                  []interface{} `json:"TransportRule"`
+	RemoteDomain                   []interface{} `json:"RemoteDomain"`
+	SafeLinksPolicy                []interface{} `json:"SafeLinksPolicy"`
+	SafeAttachmentPolicy           []interface{} `json:"SafeAttachmentPolicy"`
+	OrganizationConfig             interface{}   `json:"OrganizationConfig"`
+	AuthenticationPolicy           interface{}   `json:"AuthenticationPolicy"`
+	AntiPhishPolicy                []interface{} `json:"AntiPhishPolicy"`
+	DkimSigningConfig              interface{}   `json:"DkimSigningConfig"`
+	OwaMailboxPolicy               interface{}   `json:"OwaMailboxPolicy"`
+	AdminAuditLogConfig            interface{}   `json:"AdminAuditLogConfig"`
+	PhishFilterPolicy              []interface{} `json:"PhishFilterPolicy"`
+	// Mailbox                        []interface{}     `json:"Mailbox"`
+	AtpPolicyForO365     []interface{}     `json:"AtpPolicyForO365"`
+	SharingPolicy        []interface{}     `json:"SharingPolicy"`
+	RoleAssignmentPolicy []interface{}     `json:"RoleAssignmentPolicy"`
+	ExternalInOutlook    []*ExternalSender `json:"ExternalInOutlook"`
```

Comment:
For list items I do not understand why we use `[]*ExternalSender`, is `[]ExternalSender` not enough? If nothing is set, the list is empty anyway.